apps/sync/js/adapters/bookmarks.js

```diff
@@ -120,7 +120,9 @@ var BookmarksHelper = (() => {
     }).then(localRecord => {
       if (localRecord) {
         var newBookmark = mergeRecordsToDataStore(localRecord, remoteRecord);
-        return store.put(newBookmark, id, revisionId);
+        return store.put(newBookmark, id, revisionId).then(() => {
+          return setDataStoreId(remoteRecord.fxsyncId, id);
+        });
```

Comment:
If it's an existing record in DataStore, we don't have to update the record's matching value.
In the current Bookmarks adapter design, it is not possible to have the different FxSync ID for the two records with the same URL.
I think this should be considering with the patch of bug 1215482 together when there is a new record coming in our new bookmark store design.